Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Controller.mo

```diff
@@ -0,0 +1,2260 @@
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Measured hot water supply temperature"
+    annotation (Placement(transformation(extent={{-440,250},{-400,290}}),
+      iconTransformation(extent={{-140,210},{-100,250}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TRetPri(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Measured hot water primary return temperature"
+    annotation (Placement(transformation(extent={{-440,210},{-400,250}}),
+      iconTransformation(extent={{-140,180},{-100,220}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VHotWatPri_flow(
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate")
+    "Measured hot water primary circuit flowrate"
+    annotation (Placement(transformation(extent={{-440,170},{-400,210}}),
+      iconTransformation(extent={{-140,150},{-100,190}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput dpHotWatPri_rem[nSenPri](
+    final unit=fill("Pa", nSenPri),
+    displayUnit=fill("Pa", nSenPri),
+    final quantity=fill("PressureDifference", nSenPri)) if have_varPriPum and (have_remDPRegPri or have_locDPRegPri)
+    "Measured differential pressure between hot water supply and return in primary circuit"
+    annotation (Placement(transformation(extent={{-440,90},{-400,130}}),
+        iconTransformation(extent={{-140,90},{-100,130}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TRetSec(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature") if not have_priOnl
+    "Measured hot water secondary return temperature"
+    annotation (Placement(transformation(extent={{-440,130},{-400,170}}),
+      iconTransformation(extent={{-140,120},{-100,160}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VHotWatSec_flow(
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate") if not have_priOnl and have_secFloSen
+    "Measured hot water secondary circuit flowrate"
+    annotation (Placement(transformation(extent={{-440,-90},{-400,-50}}),
+      iconTransformation(extent={{-140,30},{-100,70}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VHotWatDec_flow(
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate") if not have_priOnl and have_varPriPum
+     and have_floRegPri and not have_secFloSen
+    "Measured hot water flowrate through decoupler leg"
+    annotation (Placement(transformation(extent={{-440,-130},{-400,-90}}),
+      iconTransformation(extent={{-140,0},{-100,40}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSupSec(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature") if not have_priOnl and
+    have_varPriPum and have_temRegPri and have_priSecTemSen
+    "Measured hot water supply temperature in secondary circuit"
```

Comment:
The Boolean condition seems incorrect: secondary HW supply temperature is a required AI point if the plant has a secondary loop, see G36 4.11.4.